server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -384,11 +396,44 @@ public void parseKnnVectorAndIndex(DocumentParserContext context, DenseVectorFie
                                 + "];"
                         );
                     }
-                    vector[index++] = (byte) value;
+                    consumer.accept((byte) value, index++);
                     squaredMagnitude += value * value;
                 }
                 fieldMapper.checkDimensionMatches(index, context);
                 checkVectorMagnitude(fieldMapper.fieldType().similarity, errorByteElementsAppender(vector), squaredMagnitude);
+                return squaredMagnitude;
+            }
+
+            private float parseHexEncodedVector(
+                DocumentParserContext context,
+                DenseVectorFieldMapper fieldMapper,
+                VectorByteDataConsumer consumer
+            ) throws IOException {
+                byte[] decodedVector = HexFormat.of().parseHex(context.parser().text());
+                fieldMapper.checkDimensionMatches(decodedVector.length, context);
+
+                byte[] vector = new byte[fieldMapper.fieldType().dims];
```

Comment:
Why do we need `vector` variable here?  What is its role? Looks like it never gets instantiated.

Reply:
Yeah, it was a leftover. Refactored as per @original-brownbear 's suggestion to remove boilerplate of consumers and just parse to a byte array.